include/odp/api/spec/comp.h

```diff
@@ -0,0 +1,752 @@
+ * and whole output has been written into output packet and corresponding
+ * result entry is updated with offset:length information and error code set to
+ * ODP_COMP_ERR_NONE.
+ *
+ * In case of partially processed array i.e. when
+ * number of packets returned < num_pkt, application should check on result
+ * on 1st failed entry (i.e. n+1th entry where
+ * n = number of entried returned) to ensure if error code is
+ * ODP_COMP_ERR_OUT_OF_SPACE.
+ *
+ * If status code on 1st failed entry set to ODP_COMP_ERR_OUT_OF_SPACE,
+ * application should call API again with corresponding output packet
+ * available with free space.
+ *
+ * All arrays should be of num_pkt size.
+ *
+ * @param         pkt_in   Input Packets
+ * @param         pkt_out  Output Packet to store output of corresponding entry
+ * @param         param    Operation parameters array
+ * @param[out]    result   Result array corresponding to each entry
+ * @param         num_pkt  Number of packets to be processed
+ *
+ * @return Number of input packets consumed (0 ... num_pkt)
+ * @retval <0 on failure
+ */
+int odp_comp_op_multi(const odp_packet_t pkt_in[],
+		      const odp_packet_t pkt_out[],
+		      odp_comp_packet_op_param_t param[],
+		      odp_comp_packet_op_result_t result[],
+		      int num_pkt);
+
+/**
+ * Comp API to process multiple packets asynchronously
+ *
+ * Performs the ASYNC decompression operations on the packet array.
+ *
+ * If session is created in ODP_COMP_ASYNC mode, result of each entry will
+ * to completion queue. Application should monitor ODP_EVENT_PACKET
+ * be queued with subtype ODP_EVENT_PACKET_COMP on queue.
+ *
+ * If session is created in ODP_COMP_SYNC mode, behavior is undefined.
+ *
+ * for compression + hash, call returns with hash appended at the end of
+ * packet on which odp_packet_op_param_t:last field is set.
+ *
+ * User should compute processed data len = output_data_range.len - digest_len,
+ * where digest_len queried through odp_comp_hash_alg_capability().
+ *
+ * For every input packet entry in in_pkt array, user should pass
+ * corresponding valid output packet handle.On return, API returns with
+ * number of entries successfully submitted for operation.
+ * In case of partially accepted array i.e. when
+ * number of packets returned < num_pkt, application my attempt to resubmit
+ * subsequent entries via calling any of the operation API.
+ *
+ * A successful submitted entry means it has been queued for processing but
+ * not fully processed. Each submitted entry results will be notified via
+ * event after operation on that entry is complete. On received event w.r.t
+ * each submitted packet, application should call odp_comp_result() to retrieve
+ * operational result.
+ *
+ * If any of the packet has status code set to ODP_COMP_ERR_OUT_OF_SPACE in
+ * its result, application should call operation API again with corresponding
+ *  output packet available with free space.
```

Comment:
Usecase: assume we enqueue 10 packets and 6 are processed OK but 7th has ODP_COMP_ERR_OUT_OF_SPACE.
1. Remaining 3 packets will not be delivered until  7th is 'resolved' (keep the order)? 
2. Application should resubmit only 7th packet, taken from the received event, with larger output packet?
3. What will happen with original 7th output packet?